src/main/scala/io/iohk/ethereum/network/FastSyncActor.scala

```diff
@@ -0,0 +1,304 @@
+package io.iohk.ethereum.network
+
+import akka.actor.{Actor, ActorLogging, ActorRef, PoisonPill, Props, Terminated}
+import akka.util.ByteString
+import io.iohk.ethereum.db.storage._
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.network.FastSyncActor._
+import io.iohk.ethereum.network.PeerActor.MessageReceived
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.network.p2p.messages.PV63._
+import io.iohk.ethereum.network.p2p.validators.ForkValidator
+import io.iohk.ethereum.utils.Config.FastSync
+import org.spongycastle.util.encoders.Hex
+
+class FastSyncActor(
+  peerActor: ActorRef,
+  blockHeadersStorage: BlockHeadersStorage,
+  blockBodiesStorage: BlockBodiesStorage,
+  receiptStorage: ReceiptStorage,
+  mptNodeStorage: MptNodeStorage,
+  evmStorage: EvmCodeStorage) extends Actor with ActorLogging {
+
+  import context.{dispatcher, system}
+
+  val GenesisBlockNumber = 0
+  val EmptyAccountStorageHash = ByteString(Hex.decode("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"))
+  val EmptyAccountEvmCodeHash = ByteString(Hex.decode("c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"))
+
+  def handleTerminated: Receive = {
+    case _: Terminated =>
+      log.info("FastSync actor terminated")
+      context stop self
```

Comment:
as I have already started working with this code, I'll let myself add some comments even thought it's still WIP.

`Terminated` is received when one of the actors you're watching (with `contet.watch`) is stopped. I don't see FastSyncActor watching any other actor. Am I missing something?